### PR TITLE
Update surgery filters

### LINKED ISSUE #388: Update filters for Surgery/PathReport

### PR DESCRIPTION
**Pull request by vsmalladi**
Close #388

### CODE REVIEW

**Comment by SenseCode** [comment]
Overall working great. Great job! @vsmalladi 
image


Surgery page:

- [ ] Missing title for `Reamings` for Surgery procedure field
- [ ] We still have IHC two fields, supposed to need this in the list?

image
image
Patient page:

- [ ] Need update surgery procedure filter the same as surgery page.
- [ ] Diagnosis date issue:
- 396,756 surgery is out of diagnosis date
- 392,698,139, missing diagnosis date.

- [ ] Y-axis cannot show full surgery name in surgery chart, not sure we have this issue before..;-)

image
image
image
image
image
image

**Comment by mingjiecn** [comment]
@vsmalladi : Can you document some details about Surgery Treatment Procedure? Not sure what is the logic for Surgery Treatment Procedure. Thank you!

**Comment by vsmalladi** [comment]
@SenseCode The Reamings should be Excision. The issue with unknown is that there is no path data associated with it. So adding unknown

i think the two IHC fields makes sense since its combination.

**Comment by vsmalladi** [comment]
@mingjiecn @SenseCode 

Logic for Surgery Treatment Procedure.

- Defaults: "Unknown

if procedure_type Not available or Ablation, keep that type

if Nephrectomy then text should be "Kidney (Nephrectomy)"

if procedure type "Biopsy" or 'Fine needle aspiration' and path type is path_biopgy then Kidney (Biopsy)
if procedure type "Excision", 'Reamings' and path type is path_metasis then Metastasis (Excision)
procedure type "Biopsy", 'Fine needle aspiratio' and path type is path_metasis then Metastasis (Biopsy)

**Comment by SenseCode** [comment]
Checked the surgery procedure missing type got fixed now ....
image

**Comment by SenseCode** [comment]
Filter works great now.
Y-axis do have the excision label showed.  
image
image

File: src/encoded/types/surgery.py [hunk 1]
```diff
@@ -111,6 +111,71 @@ def pathology_report(self, request, pathology_report):
...
+
+
+    @calculated_property(define=True, schema={
+            "title": "Surgery Diagnosis Procedure",
+            "type": "string",
+        })
+    def surgery_diagnosis(self, request, procedure_type, pathology_report):
+        surgery_diagnosis = None
+        if procedure_type in ["Not available"]:
+            surgery_treatment = None
```
Comment (mingjiecn) [inline src/encoded/types/surgery.py]: Should surgery_treatment  be surgery_diagnosis?

Comment (vsmalladi) [reply src/encoded/types/surgery.py]: Fixed variable name

**Comment by mingjiecn** [comment]
I am concerned about the procedure type "not available". For the real data, do we have surgery with surgery procedure that is not available? If so what is the logic here to put procedure type "not available" into treatment, not into diagnosis.

**Comment by vsmalladi** [comment]
We shouldn’t this was just logic to show errors and we can catch it. I can put it in diagnosis as well

**Comment by vsmalladi** [comment]
@mingjiecn @SenseCode 

For diagnosis date found the issue.

- 396 has no confirmed pathology so this is why the diagnosis date is different than Nephrectomy
- 756 only has an ablation which isn't confirmed pathology diagnosis (potential for non-cancer
- 392,698,139 has no confirmed pathology so this is why the diagnosis date is different than Nephrectomy 

I think this is an error in test data for the Nephrectomy issue. 

For the ablation we should ask Alana today

**Comment by SenseCode** [comment]
- [ ] What's this?

image

- [x] Fixed Y-axis cannot show all labels issue
image

**Comment by vsmalladi** [comment]
> * [x]  What's this?
> 
> image

This surgery has Excision but no pathology report for confirmation. So won't display the surgery in timeline

**Review by mingjiecn — APPROVED** [review]

**Comment by SenseCode** [comment]
> > * [x]   What's this?
> > 
> > image
Just kind of weird with surgery chart but shows no surgery....

**Review by SenseCode — APPROVED** [review]
Works great , great job! @vsmalladi

**Pull request merged**